client/state/themes/actions.js

```diff
@@ -404,6 +405,44 @@ export function clearActivated( siteId ) {
 }
 
 /**
+ * Triggers a network request to install WordPress.com theme on Jetpack site.
+ * After installataion it switches page to the customizer
+ * Requires Jetpack 4.4
+ *
+ * @param  {String}   themeId      WP.com Theme ID
+ * @param  {String}   siteId       Jetpack Site ID
+ * @return {Function}              Action thunk
+ */
+export function installAndTryAndCustomize( themeId, siteId ) {
+	return ( dispatch, getState ) => {
+		return dispatch( installTheme( themeId, siteId ) )
+			.then( () => {
+				const theme = getTheme( getState(), siteId, themeId );
+				if ( ! theme ) {
+					return;
+				}
+				const url = getThemeCustomizeUrl( getState(), theme, siteId );
+				page( url );
```

Comment:
If you want to modularize things, you could put the contents of the `.then()` statement into a separate action. That might simplify testing a bit.